diff --git a/node_modules/@vue/babel-plugin-jsx/dist/transform-vue-jsx.js b/node_modules/@vue/babel-plugin-jsx/dist/transform-vue-jsx.js
index 66ee202..48bd9a3 100644
--- a/node_modules/@vue/babel-plugin-jsx/dist/transform-vue-jsx.js
+++ b/node_modules/@vue/babel-plugin-jsx/dist/transform-vue-jsx.js
@@ -156,7 +156,18 @@ exports.transformJSXElement = transformJSXElement;
 exports.default = ({
     JSXElement: {
         exit(path, state) {
+            const { file } = path.hub;
+            const { filename } = file.opts;
+            if (state === undefined && exports.state) {
+              state = exports.state
+              state.file = file
+              state.filename = filename
+            }
+            if (exports.state == undefined && state) {
+              exports.state = state;
+            }
             path.replaceWith(transformJSXElement(path, state));
         },
     },
 });
+exports.state = undefined
